internal/queue/queue.go

```diff
@@ -61,8 +61,16 @@ type Queue struct {
 
 	// wakeup
 	wakeupCh chan struct{}
+
+	retryFunc ShouldRetryFunc
 }
 
+// ShouldRetryFunc is passed a key, the number of attempts that this item has already had, and the (potentially wrapped)
+// error from the queue handler. If the client should retry, the error value should be nil. Optionally, the function
+// may return a value for the delay, to override the default rate limit backoff. The returned error is bubbled up,
+// and can just be the errors that was passed in, if the function does not care to decorate the error.
+type ShouldRetryFunc func(ctx context.Context, key string, timesTried int, originallyAdded time.Time, err error) (*time.Duration, error)
```

Comment:
Can you explain the usage of the duration pointer instead of value?

Reply:
I think this explains it. Or should I be explicit about nil vs. 0 value?

> Optionally, the function
> may return a value for the delay, to override the default rate limit backoff.